### PR TITLE
Backport: Upgrade h2 to version 2.0.206

### PR DESCRIPTION
**Pull request by cocreature**
backport from #12307 
changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: maven_install_2.12.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "dependency_tree": {
-        "__AUTOGENERATED_FILE_DO_NOT_MODIFY_THIS_FILE_MANUALLY": -429398716,
+        "__AUTOGENERATED_FILE_DO_NOT_MODIFY_THIS_FILE_MANUALLY": -810309042,
```
Comment (cocreature) [inline maven_install_2.12.json]: I just repinned those but apparently the 1.18 branches haven’t been modified sufficiently long that this turned into a huge change. not much we can do about this.

File: bazel-java-deps.bzl [hunk 2]
```diff
@@ -44,7 +44,7 @@ netty_version = "4.1.67.Final"
 # <tcnative.version>2.0.42.Final</tcnative.version>
 # ```
 netty_tcnative_version = "2.0.40.Final"
-grpc_version = "1.41.0"
+grpc_version = "1.35.0"
```
Comment (cocreature) [inline bazel-java-deps.bzl:47]: See https://github.com/digital-asset/daml/pull/11748/files#r751503832, this just reverts stuff back to what we were already using.

File: maven_install_2.12.json [hunk 3]
```diff
@@ -62,15 +62,15 @@
             {
                 "coord": "co.fs2:fs2-core_2.12:2.5.6",
                 "dependencies": [
-                    "org.scala-lang:scala-library:2.12.14",
                     "org.typelevel:cats-effect_2.12:2.5.1",
+                    "org.scala-lang:scala-library:2.12.15",
```
Comment (nmarton-da) [inline maven_install_2.12.json]: is this needed?

**Review by nmarton-da — APPROVED** [review]
Looks good, but can't comment on the scala lib version changes + dependency bumps. If that is needed, and CI passes, I guess it is okay.

File: maven_install_2.12.json (hunk 3 above)
Comment (cocreature) [reply maven_install_2.12.json]: this is a generated file and we need to regenerate it. I’m not quite sure what determines these versions but we cannot change them beyond regenerating.

**Comment by cocreature** [comment]
turning into a draft until the postgres issues are resolved.

File: ledger/participant-integration-api/src/main/resources/db/migration/h2database-appendonly/V1__Append_only_schema.sql [hunk 4]
```diff
@@ -24,7 +24,7 @@ CREATE TABLE configuration_entries (
     recorded_at BIGINT NOT NULL,
     submission_id VARCHAR NOT NULL,
     typ VARCHAR NOT NULL,
-    configuration BYTEA NOT NULL,
+    configuration BINARY LARGE OBJECT NOT NULL,
```
Comment (nmarton-da) [inline ledger/participant-integration-api/src/main/resources/db/migration/h2database-appendonly/V1__Append_only_schema.sql:27]: apparently H2 changed behavior with original BYTEA (boiling down to BINARY VARYING): limited to 1Mb. To erase this limit we switch to BINARY LARGE OBJECT everywhere needed

File: ledger/participant-integration-api/src/main/scala/platform/store/backend/h2/H2Field.scala [hunk 5]
```diff
@@ -17,3 +19,15 @@ private[h2] case class IntArrayOptional[FROM](
   override def convert: Option[Iterable[Int]] => Array[java.lang.Integer] =
     _.map(_.view.map(Int.box).toArray).orNull
 }
+
+private[h2] case class H2Bytea[FROM](extract: StringInterning => FROM => Array[Byte])
+    extends Field[FROM, Array[Byte], InputStream] {
+  override def convert: Array[Byte] => InputStream = new ByteArrayInputStream(_)
```
Comment (nmarton-da) [inline ledger/participant-integration-api/src/main/scala/platform/store/backend/h2/H2Field.scala:25]: For H2 JDBC we need InputStream, so it is recognized as a BINARY LARGE OBJECT

**Review by mziolekda — APPROVED** [review]

**Pull request merged**